### PR TITLE
MAUI-743: Specify error row for lab import

### PR DESCRIPTION
**Pull request by chris-pollard**
### Changes:

- Throw upload error when database returns no result on entity id.

### CODE REVIEW

File: packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js [hunk 1]
```diff
@@ -29,7 +29,15 @@ const createWorkBookParser = () => {
 const createImporter = models => {
   const constructResponseExtractor = () => async input => {
     const entityCode = input[ENTITY_CODE_KEY];
-    const { id: entityId } = await models.entity.findOne({ code: entityCode });
+    const getEntityId = async () => {
+      try {
+        const { id } = await models.entity.findOne({ code: entityCode });
+        return await id;
+      } catch {
+        throw new UploadError({ message: `No Entity Id found with code '${entityCode}'` });
```
Comment (edmofro) [inline packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js:37]:
```suggestion
        throw new UploadError({ message: `No entity found with code '${entityCode}'` });
```

File: packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js [hunk 2]
```diff
@@ -29,7 +29,15 @@ const createWorkBookParser = () => {
 const createImporter = models => {
   const constructResponseExtractor = () => async input => {
     const entityCode = input[ENTITY_CODE_KEY];
-    const { id: entityId } = await models.entity.findOne({ code: entityCode });
+    const getEntityId = async () => {
+      try {
+        const { id } = await models.entity.findOne({ code: entityCode });
+        return await id;
```
Comment (edmofro) [inline packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js:35]: No need to `await` the id here :-) 
```suggestion
        return id;
```

File: packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js [hunk 3]
```diff
@@ -54,7 +62,7 @@ const getEntitiesGroupedBySurveyCode = async (models, inputsPerSurvey) => {
 
     const survey = surveys.find(survey => survey.name === surveyName);
     if (!survey) {
-      throw new Error(`No survey found with name '${surveyName}'`);
+      throw new UploadError({ message: `No survey found with name '${surveyName}'` });
```
Comment (edmofro) [inline packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js:65]: Nice change

**Review by edmofro — CHANGES_REQUESTED** [review]

File: packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js (hunk 2 above)
Comment (chris-pollard) [reply packages/meditrak-server/src/apiV2/import/importStriveLabResults/importStriveLabResults.js]: Thanks @edmofro . Yes I wasn't sure about whether await was required inside the try/catch promise structure.

**Review by edmofro — APPROVED** [review]
:+1:

**Pull request merged**